src/Client.h

```diff
@@ -140,21 +141,48 @@ class Client : public ClientBase{
 		int serverNo = 0;
 
 		TicksTime rxTime;
-
-		ret = msg_recvfrom(ifd,
-				           g_fds_array[ifd]->recv.cur_addr + g_fds_array[ifd]->recv.cur_offset,
-				           g_fds_array[ifd]->recv.cur_size,
-				           &recvfrom_addr);
-		if (ret == RET_SOCKET_SHUTDOWN) {
-			if (g_fds_array[ifd]->sock_type == SOCK_STREAM) {
-				exit_with_log("A connection was forcibly closed by a peer",SOCKPERF_ERR_SOCKET,g_fds_array[ifd]);
+#ifdef  USING_VMA_EXTRA_API
+		if (!g_vma_api || !g_pApp->m_const_params.is_vmapoll)
+#endif
+		{
+			ret = msg_recvfrom(ifd,
+					   g_fds_array[ifd]->recv.cur_addr + g_fds_array[ifd]->recv.cur_offset,
+					   g_fds_array[ifd]->recv.cur_size,
+					   &recvfrom_addr);
+			if (ret == RET_SOCKET_SHUTDOWN) {
+				if (g_fds_array[ifd]->sock_type == SOCK_STREAM) {
+					exit_with_log("A connection was forcibly closed by a peer",SOCKPERF_ERR_SOCKET,g_fds_array[ifd]);
+				}
 			}
+			if (ret < 0) return 0;
 		}
-		if (ret < 0) return 0;
+#ifdef USING_VMA_EXTRA_API
+		int nbytes = ret;
+		vma_buff_t* tmp_vma_poll_buff = g_vma_poll_buff;
+		while(tmp_vma_poll_buff || nbytes) {
+			if (tmp_vma_poll_buff && !nbytes){
+				if (g_fds_array[ifd]->recv.cur_offset) {
+					g_fds_array[ifd]->recv.cur_addr = g_fds_array[ifd]->recv.buf;
+					memcpy(g_fds_array[ifd]->recv.cur_addr + g_fds_array[ifd]->recv.cur_offset,(uint8_t*)tmp_vma_poll_buff->payload, tmp_vma_poll_buff->len);
+				}
+				else {
+					g_fds_array[ifd]->recv.cur_addr = (uint8_t*)tmp_vma_poll_buff->payload;
+				}
+				ret = tmp_vma_poll_buff->len;
+				recvfrom_addr = g_vma_comps->src;
+				if (ret == RET_SOCKET_SHUTDOWN) {
+					if (g_fds_array[ifd]->sock_type == SOCK_STREAM) {
+						exit_with_log("A connection was forcibly closed by a peer",SOCKPERF_ERR_SOCKET,g_fds_array[ifd]);
+					}
+				}
+				if (ret < 0) return 0;
+				nbytes = ret;
+			}
 
```

Comment:
@AlaAyesh As far as sockperf is used to demonstrate performance results we should select a way with better performance. Have had a chance to compare performance for these solutions? In any way let use #1 option adding comment describing possible ways for implementation and reason of our choice.